ajax_with_jquery_exercise/nacker_hews_AJAX/app.js

```diff
@@ -0,0 +1,292 @@
+          });
+          $star.data("story_id",value.story_id);
+          $star.data("favorite_id",value.id);
+          let $text = $("<a>", {
+            class: "topic-text",
+            target: "blank",
+            href: value.url,
+            text: value.title
+          });
+          let $link = $("<span>", {
+            class: "class-filter",
+            text: `(${value.url})`
+          });
+          $li.append($star,$text,$link);
+          $topiclist.append($li);
+          });
+          hackerNews = [];
+
+        })
+  }
+
+  //Slide down forms
+  $login.on("click", function(event) {
+    $(".form-row").slideToggle(300);
+  })
+
+
+  //Top stories click
+  $topStories.on("click", function(event) {
+    fetchTopNews();
+  })
+
+  //Best stories click
+  $bestStories.on("click", function(event) {
+    fetchTopNews();
+  })
+
+  //Best stories click
+  $newStories.on("click", function(event) {
+    fetchTopNews();
+  })
+
+
+  //Toggle Icon code
+  $topiclist.on("click", ".glyphicon", function(event) {
+    console.log($(event.target));
+    if ($(event.target)[0].className === "glyphicon glyphicon-star-empty") {
+      //Post story
+      $(event.target).toggleClass("glyphicon-star-empty");
+      $(event.target).toggleClass("glyphicon-star");
+      $.get("https://hacker-news.firebaseio.com/v0/item/" + $(event.target).data().story_id + ".json?print=pretty")
+      .then(function(res){
+        $.ajax({
+          type: "Post",
+          url: "https://hn-favorites.herokuapp.com/stories.json",
+          headers: {
+            authorization: JSON.parse(localStorage.getItem("Nacker"))[0]
+          },
+          data: JSON.stringify({
+            hacker_news_story: {
+              story_id: res.id,
+              by: res.by,
+              title: res.title,
+              url: res.url
+            }
+          }),
+          contentType: "application/json"
+        }).then(function(res){
+          $(event.target).data("favorite_id", res.id);
+          console.log(res);
+        }).catch(function(res){
+          console.log(res);
+        })
+      });
+    } else if ($(event.target)[0].className === "glyphicon glyphicon-star") {
+      //Delete story
+      $(event.target).toggleClass("glyphicon-star-empty");
+      $(event.target).toggleClass("glyphicon-star");
+      $.ajax({
+        type: "Delete",
+          url: "https://hn-favorites.herokuapp.com/stories/" + $(event.target).data().favorite_id +".json",
+          headers: {
+            authorization: JSON.parse(localStorage.getItem("Nacker"))[0]
+          }
+      }).then(function(res){
+        console.log(res);
+      })
+    };
+  });
+
+  //Login for user
+  $loginForm.on("submit",function(event) {
+    event.preventDefault();
+    $.post("https://hn-favorites.herokuapp.com/login", {
+      email: $("#username").val(),
+      password: $("#password").val()
+    }).then(function(res){
+      alert("Log in successful!");
+      storage.push(res.auth_token);
```

Comment:
I'm not super clear on the point of this `storage` array. Why are you doing this in addition to using `localStorage`?